web/index.js

```diff
@@ -221,14 +218,52 @@ function handleData(data) {
 /**
  * When a refocus.bot.actions is dispatch it is handled here.
  *
- * @param {BotAction} action - Bot Action object that was dispatched
+ * @param {Object} action - Bot Action object that was dispatched
  */
 function handleActions(action) {
   if (action.detail.roomId === roomId) {
     bdk.log.debug('Action Received: ', action.detail);
   }
 }
 
+/**
+ * @param {Object} data - request
+ * @param {Object} rolesFromSettings - from RoomType
+ */
+function createBotRoles(data, rolesFromSettings) {
+  const rolesBotData = data.body.filter((bd) => bd.name ===
+    'participantsBotRoles')[ZERO];
+  roles = rolesBotData ? JSON.parse(rolesBotData.value) : rolesFromSettings;
+  if (rolesBotData) {
+    rolesBotDataId = rolesBotData.id;
+  } else {
+    /* eslint-disable no-return-assign */
+    bdk.createBotData(roomId, botName,
+      'participantsBotRoles', serialize(rolesFromSettings))
+      .then((res) => rolesBotDataId = res.body.id);
+  }
+
+  roles.forEach((role) => {
+    currentRole[role.label] = data.body
+      .filter((bd) => bd.name === 'participants' + role.label)[ZERO];
+  });
+}
+
+/**
+ * Initiate an empty bot data that will contains all the participants in the room
+ * if it's not created yet.
+ *
+ * @param {Array} listOfBotData - all the bot data for the room & participant-bot
+ */
+function assignParticipants(listOfBotData) {
+  if (!listOfBotData) return;
+  const ASSIGNED = 'assignedParticipants';
```

Comment:
why all caps?

Reply:
I have this as a convention but we don't use generally, will lower it

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

https://www.javascripttutorial.net/es6/javascript-const/#:~:text=Introduction%20to%20the%20JavaScript%20const%20keyword&text=By%20convention%2C%20the%20constant%20identifiers,values%20can't%20be%20reassigned.&text=However%2C%20variables%20created%20by%20the%20const%20keyword%20are%20immutable.

Reply:
It's perfectly ok to have constants with all uppercase IMO